app/controllers/pieces_controller.rb

```diff
@@ -17,6 +21,10 @@ def show
 
   def edit
     @piece = Piece.find(params[:id])
+    unless current_user.can_edit?(@piece)
+      flash[:notice] = "You can only edit a piece you created"
+      redirect_to piece_path(@piece)
+    end
```

Comment:
Put lines 24-27 in a method and put it in this controller. The create a controller callback in this controller for only this action.